internal/npm_install/generate_build_file.ts

```diff
@@ -1009,13 +1012,15 @@ js_library(
 # Target is used as dep for main targets to prevent circular dependencies errors
 js_library(
     name = "${pkg._name}__contents",
+    npm_package = True,
     srcs = [":${pkg._name}__files", ":${pkg._name}__nested_node_modules"],${namedSourcesStarlark}
     visibility = ["//:__subpackages__"],
 )
 
 # Typings files that are part of the npm package not including nested node_modules
 js_library(
-    name = "${pkg._name}__typings",${dtsStarlark}
+    name = "${pkg._name}__typings",
+    npm_package = True,${dtsStarlark}
```

Comment:
seeing this API seems a little wrong. If I'm making a `js_library` in my own code why wouldn't I say it's an npm package?
Should it be named differently? if users aren't meant to call it, could be a longer name like `third_party_npm_package` 
Another possibility is a different rule name like external_js_library ("external" is nice echo from rules_jvm_external) - it would be implemented in the same file with js_library so they share code

Reply:
Discussed offline & renamed to external_npm_package. Also renamed related provided to ExternalNpmPackageInfo to match.